src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -531,8 +531,9 @@ public bool HasUnsavedChanges
         /// Returns if current workspace is readonly.
         /// </summary>
         public bool IsReadOnly
-        {
-            get { return isReadOnly; }
+        {   
+            //if the workspace contains xmlDummyNodes it's effectively a readonly graph.
+            get { return isReadOnly || this.containsXmlDummyNodes(); }
```

Comment:
Hmm.. I'm trying to see if we could move this.containsXmlDummyNodes() to `WorkspaceModel` constructor, do we need to check it all the times for saving?

Reply:
@QilongTang yeah, we want this to be recomputed whenever there readonly gets checked during different save pathways - if someone removes the nodes from a graph we should let them save over the original file for instance.